tawazi/_dag/dag.py

```diff
@@ -508,89 +657,260 @@ def __call__(self, *args: P.args, **kwargs: P.kwargs) -> RVDAG:
+                    f"currently DAG does not support keyword arguments: {kwargs}"
+                )
+
+        if description_context:
+            logger.warning("Describing SubDAG {} in DAG", self)
+
+            # NOTE: can't call the base describing function because composed DAGs can't be supported in that case
+            #  so must modify ExecNodes of SubDAG
+            node.DAG_PREFIX.append(self.qualname)
+
+            def to_subdag_id(id_: str) -> str:
+                return ".".join(node.DAG_PREFIX + [id_])
+
+            # only the ExecNodes of the SubDAG must be affected by the is_active
+            is_active = False if ARG_NAME_ACTIVATE not in kwargs else kwargs[ARG_NAME_ACTIVATE]
+
+            input_uxns = [UsageExecNode(to_subdag_id(uxn.id), uxn.key) for uxn in self.input_uxns]
+
+            # provided args to the subdag
+            arg_uxns = construct_subdag_arg_uxns(
+                *args, to_subdag_id=to_subdag_id, qualname=self.qualname
+            )
+
+            registered_input_ids: List[str] = []
+            # provided *args to the call is <= than input_uxns! because of defaults args
+            for axn, uxn in zip(arg_uxns, input_uxns):  # strict=False
+                # a stub that fills the value of an input ExecNode with an arg of the subdag
+                stub: LazyExecNode[[UsageExecNode], UsageExecNode] = LazyExecNode(
+                    id_=uxn.id,
+                    exec_function=lambda x: x,
+                    resource=consts.Resource.main_thread,
+                    args=[axn],
+                    # during subdag construction,
+                    # there are only two frames to get to the actual call site
+                    call_location_frame=2,
+                )
+                # register this LazyExecNode in the dict
+                # pass kwargs to pass in the twz_active!
+                _val: UsageExecNode = stub(axn, **kwargs)
+                registered_input_ids.append(uxn.id)
+
+            # updating ids of results already registered in the DAG due to pipeline.setup and default args
+            node.results.update(
+                StrictDict((to_subdag_id(id_), res) for id_, res in self.results.items())
+            )
+
+            # updating values of the ExecNodes with the new Ids only for the inputs that were changed!
+            graph = DiGraphEx()
+            for xn in self.exec_nodes.values():
+                graph.add_exec_node(xn)
+            # must go by order because Dict doesn't respect the order of insertion
+            while len(graph):
+                id_ = graph.remove_any_root_node()
+                exec_node = self.exec_nodes[id_]
+                new_id = to_subdag_id(id_)
+                # input ExecNode was already registered during step for zip
+                if new_id in registered_input_ids:
+                    logger.debug(
+                        "Skipping ExecNode {} because it is an already registered input", new_id
```

Comment:
it is already a registered output rather

Reply:
no it is a registered input of the subdag

Reply:
oups sorry typo on me behalf, I meant "because it is already a registered input"

Reply:
done